### ISSUE TITLE
Rectangles displayed instead of numbers of commits behind

### ISSUE THREAD

**Issue by mgryszko** [JanDeDobbeleer/oh-my-posh#145]
### Prerequisites

- [x] I have read and understand the `CONTRIBUTING` guide
- [x] I looked for duplicate issues before submitting this one

### Description

Some non-Powerline symbols are not rendered correctly. Used font: Cascadia Mono PL. 

Compare oh-my-posh 2.0.487 on Windows PowerShell 5.1:

![image](https://user-images.githubusercontent.com/516050/98914257-b98cbe80-24c8-11eb-9086-d869194666e5.png)

with the oh-my-post 3.33.0-beta on PowerShell Core 7.0.3:

![image](https://user-images.githubusercontent.com/516050/98914040-7a5e6d80-24c8-11eb-8002-d5fe3e52169e.png)

### Environment

- Oh my Posh version: 3.33.0-beta
- Theme: fish
- Operating System: Windows 1909 build 18363.1139
- Shell: PowerShell Core 7.0.3
- Terminal: Windows Terminal 1.3.2651.0

### Steps to Reproduce

`Set-PoshPrompt -Theme fish`

**Expected behavior:** look&feel of oh-my-posh 2

**Actual behavior:** rectangles displayed instead of commits behind

**Comment by PhilipHaglund** [comment]
Can confirm I have the same issue.
### Environment

- Oh my Posh version: 3.33.0-beta
- Theme: powerlevel10k_classic
- Operating System: Windows 20H2 (19042.630)
- Shell: PowerShell 7.1
- Terminal: Windows Terminal 1.4.3141.0

**Comment by JanDeDobbeleer** [comment]
You need to set a Nerd Font in your terminal. See [here](https://ohmyposh.dev/docs/installation#font) for more context.

**Comment by johannesg** [comment]
It does not seem to work for all Nerd Fonts. I tried Cascayda Cove (NF version of Cascadia Code, and my preferred font), Fira Code and Meslo. Out of those 3, only Meslo worked for me.

**Comment by JanDeDobbeleer** [comment]
They should be universally equal when it comes to those things but I'll double check!

**Comment by PhilipHaglund** [comment]
Found out that I've installed the Mono-version and after the latest version of Oh my Posh it broke.
Just installing the the Non-Mono version made everything work for me.
![image](https://user-images.githubusercontent.com/14297750/98937756-29aa3d00-24e7-11eb-88c9-c465fe972ded.png)

Terminal - settings.json
![image](https://user-images.githubusercontent.com/14297750/98937904-637b4380-24e7-11eb-8b98-0c670baa3ef3.png)

![image](https://user-images.githubusercontent.com/14297750/98937972-7beb5e00-24e7-11eb-855e-db0413e8f038.png)

**Comment by johannesg** [comment]
Unfortunately that didn't work for me. I uninstalled all CaskaydiaCove fonts and reinstalled the windows compatible version. I still get no symbols.

![image](https://user-images.githubusercontent.com/186378/98940022-9e32ab00-24ea-11eb-9433-56a62ee76c0e.png)

**Comment by JanDeDobbeleer** [comment]
That's really weird. Are you sure the font that's set is correct? Because even the powerline symbol is broken, that one I haven't seen before.

**Comment by mgryszko** [comment]
I installed the same font as @PhilipHaglund is using (BTW, very nice font) and I'm still having the same issue.

**Comment by JanDeDobbeleer** [comment]
Oh my Posh has no way to influence the font and uses the codes in the official Nerd Font cheat sheet. The character encoding of Powershell shouldn't influence this either (at least not with that output), although, as a test, does setting that to UTF8 change anything?

**Comment by nwykes** [comment]
> I installed the same font as @PhilipHaglund is using (BTW, very nice font) and I'm still having the same issue.

You've configured your Terminal to use that font?

**Comment by JanDeDobbeleer** [comment]
@mgryszko this is that font in my terminal:
<img width="618" alt="afbeelding" src="https://user-images.githubusercontent.com/2492783/98987517-92fd7080-2526-11eb-9e96-b07eda56a0fe.png">
I would really check the settings because 9/10 something is off in your terminal configuration.

**Comment by cprivitere** [comment]
Can confirm, using the microsoft provided PL version of Cascadia Code (it's bundled in their overall Cascadia Code release) https://github.com/microsoft/cascadia-code and version 3 of oh-my-posh does result in boxes for the git commit version numbers.  Version 2 didn't do this.

There is literally no other config to check. This is stock Windows Terminal (preview version in case that matters) with a from scratch settings.json with only the font choice specified. Even a fresh install of Powershell 7.1, but it happnes on 7.0 too. Did this on two separate machines even.

Using oh-my-posh3 installed via `Install-Module oh-my-posh -Scope CurrentUser -AllowPrerelease`

What sort of useful info can we get to help debug this? Is there some possibility that the symbol being called in 3 is different than 2 and the font some people use doesn't have one of them? 

Here's it not working in 3:
![image](https://user-images.githubusercontent.com/23177737/99026677-250a7680-2531-11eb-933c-37cd7f95067d.png)

Here's it working in 2, the only change is oh-my-posh:
![image](https://user-images.githubusercontent.com/23177737/99026933-af52da80-2531-11eb-895f-c5886873ec2f.png)

**Comment by JanDeDobbeleer** [comment]
OK, so, there's nothing to debug in that case. The only thing needed is to create a config that uses the icons available in Cascadia Code. This is not a bug, it's actually by design as I use Nerd Fonts to have a wider range of icons. When using the nerd font version of Cascadia Code, I do get all correct symbols in my terminal, which is as expected. 

You can dump the current config with `Write-PoshTheme`, store it in a file, adjust the icons and use that file when setting the prompt using `Set-PoshTheme`.

For these icons specifically, using Cascadia Code PL, these are the [git icons](https://ohmyposh.dev/docs/git#properties) which needs adjusting:

```json
"branch_identical_icon": "\u2261 ",
"branch_ahead_icon": "\u2191 ",
"branch_behind_icon": "\u2193 ",
"branch_gone_icon": "\u2262 "
```

There will be other icons which don't map but hopefully this provides the information you were looking for.

EDIT: looking at this I tried these glyphs in Nerd Font and they also work there. While this won't fix all icons (some are simply not available in Cascadia Code PL), I'll create a small adjustment that maps as many of these for the Git segment.

**Comment by JanDeDobbeleer** [comment]
I'm going to close this as I can't reproduce this with any NF set correctly and did my best to support Cascadia Code PL as much as I could. Let me know if there's anything else I can do.

**Comment by cprivitere** [comment]
That fixed it on my end! Thanks!

**Comment by PhilipHaglund** [comment]
Just FYI, my problem was related to https://github.com/microsoft/terminal/issues/3257.
So just installing the fonts system-wide fixed my issue.
https://stackoverflow.com/questions/55264642/how-to-force-win10-to-install-fonts-in-c-windows-fonts

**Comment by KyleKorndoerfer** [comment]
I just updated to oh-my-posh 3.53.1 beta and am having the same issues within Windows Terminal. I have installed different NF fonts using the 'install for all users' option but only Meslo seems to work (which is not my preferred font). @PhilipHaglund what worked for you (including any changes to terminal, encoding, etc.)? I had everything working great in v2 and really regretting moving to v3 at the moment.

**Comment by PhilipHaglund** [comment]
> 
> 
> I just updated to oh-my-posh 3.53.1 beta and am having the same issues within Windows Terminal. I have installed different NF fonts using the 'install for all users' option but only Meslo seems to work (which is not my preferred font). @PhilipHaglund what worked for you (including any changes to terminal, encoding, etc.)? I had everything working great in v2 and really regretting moving to v3 at the moment.

I just installed "CaskaydiaCove NF" as Administrator to "All Users".
Do you still have the same issue with boxes? Are you using the latest Terminal?

**Comment by johannesg** [comment]
> I just updated to oh-my-posh 3.53.1 beta and am having the same issues within Windows Terminal. I have installed different NF fonts using the 'install for all users' option but only Meslo seems to work (which is not my preferred font). @PhilipHaglund what worked for you (including any changes to terminal, encoding, etc.)? I had everything working great in v2 and really regretting moving to v3 at the moment.

I also have the same problem still. Meslo works always, but Cascaidya does not work, whether it's installed for all users or not.

Version: 3.53.1
Theme: Paradox
Terminal version: 1.4.3243.0

FontFace: CascadyaCode NF (Installed for all users)
![image](https://user-images.githubusercontent.com/186378/102322012-b4ff6e00-3f7e-11eb-8d1c-77ae3eea5e16.png)

FontFace: MesloLGM NF (Installed for current user)
![image](https://user-images.githubusercontent.com/186378/102322074-c8aad480-3f7e-11eb-8ebf-9ae3dfc7eb11.png)

FontFace: Cascadia Mono PL (Installed for current user)
![image](https://user-images.githubusercontent.com/186378/102322105-d4969680-3f7e-11eb-9a47-b7d418fcbd4b.png)

**Comment by johannesg** [comment]
Oh s**t. I mispelled the font name, it should of course be "CaskaydiaCove NF". Now it works. Ignore my previous comment :-)

**Comment by KyleKorndoerfer** [comment]
I must have made the same mistake as it is working for me now as well. Only issue remaining for me is getting GIT to change colors on status as it did in v2!

**Comment by JanDeDobbeleer** [comment]
@KyleKorndoerfer that's a feature, check the [git segment docs](https://ohmyposh.dev/docs/git), it should all be in there!

**Comment by KyleKorndoerfer** [comment]
@JanDeDobbeleer I got it after poking at it for a bit. What confused me initially was the reference to [color](https://ohmyposh.dev/docs/configure#colors) for the various git states so I was using the angle bracket syntax instead of straight color numbers. All good now, thanks!

**Comment by jarmos-san** [comment]
It appears Windows users are a bit confused on how to get the glyphs working. The [documentation](https://ohmyposh.dev/docs/installation) helps a lot to install the OMP v3. So ensure you've it installed first, then follow these steps to get the fonts working.

- [Download](https://www.nerdfonts.com/font-downloads) your preferred font family & extract the zipped file.
- Right click on the specific font face & **"Install for all users"**. Check the GIF for clarification where I'm using the Meslo LG L font face.

![fhZausSWWU](https://user-images.githubusercontent.com/31373860/104754330-19c90680-577f-11eb-97d5-9624337b68cb.gif)

- Configure Windows Terminal (if you're not using it, you should) to use the specific font face. Meslo LG L in my case.

```json
    "profiles": {
        "defaults": {
            "fontFace": "MesloLGL Nerd Font",
```

Enjoy!

**Comment by hendripermana** [comment]
> @mgryszko this is that font in my terminal:
> <img alt="afbeelding" width="618" src="https://user-images.githubusercontent.com/2492783/98987517-92fd7080-2526-11eb-9e96-b07eda56a0fe.png">
> I would really check the settings because 9/10 something is off in your terminal configuration.

what theme you use

**Comment by JanDeDobbeleer** [comment]
@hendripermana that's `jandedobbeleer`

**Comment by nayanex** [comment]
> You need to set a Nerd Font in your terminal. See [here](https://ohmyposh.dev/docs/installation#font) for more context.

The link is broken

**Comment by JanDeDobbeleer** [comment]
@nayanex see the [docs](https://ohmyposh.dev/docs/fonts) for the latest information.

**Comment by deadcoder0904** [comment]
> ```json
> "branch_identical_icon": "\u2261 ",
> "branch_ahead_icon": "\u2191 ",
> "branch_behind_icon": "\u2193 ",
> "branch_gone_icon": "\u2262 "
> ```

@JanDeDobbeleer Where do I put these? I am using **Cascadia Code PL** & it looks like these icons using Git Branch are the only thing that is not working now. Everything else just works fine.

**Comment by JanDeDobbeleer** [comment]
@deadcoder0904 you can have a look at the [hotstick](https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/themes/hotstick.minimal.omp.json#L33) theme for reference.

**Comment by deadcoder0904** [comment]
@JanDeDobbeleer no I mean where does that file exist in my system?

**Comment by JanDeDobbeleer** [comment]
@deadcoder0904 you want to [export the config](https://ohmyposh.dev/docs/pwsh#override-the-theme-settings) and adjust it so you have something that stays changed as oh-my-posh gets updated.

**Comment by luisfcevallos** [comment]
> You need to set a Nerd Font in your terminal. See [here](https://ohmyposh.dev/docs/installation#font) for more context.

Thank you, this link works: https://github.com/ryanoasis/nerd-fonts/releases/tag/v2.1.0

**Comment by deadcoder0904** [comment]
@luisfcevallos Did you manage to get the icons working? I can't get it to work using [lsd](https://github.com/Peltoche/lsd). Even Github commits don't have an icon:

![lsd](https://user-images.githubusercontent.com/16436270/127266940-4c9910d7-0884-4bcd-b695-ea2a6cbaeaca.png)

**Comment by JanDeDobbeleer** [comment]
@deadcoder0904 did you have a look at the [font guide](https://ohmyposh.dev/docs/fonts)?

**Comment by deadcoder0904** [comment]
@JanDeDobbeleer Yes & I don't know which font to install as there are so many & no one with the name `MesloLGM NF`. I did install `Meslo LG M` that has `Windows Compatible` beside it. Followed countless blogpost the last time but didn't get the icons to work so I gave up eventually.

Ideally, an automated script would've been good to install fonts.

**Comment by JanDeDobbeleer** [comment]
@deadcoder0904 font installs are a machine/terminal thing and not related in any way to oh-my-posh's functionality. The only thing I see is that your terminal is not properly configured, the font could be installed the right way, maybe the name of the font you set is wrong.

**Comment by deadcoder0904** [comment]
@JanDeDobbeleer No, having done this countless times on Linux & Mac since 2015 & now Windows, I don't think its wrong as it changes the font perfectly fine.

Here's my config:

```JSON
"profiles": {
    "defaults": {
      "startingDirectory": "%__CD__%",
      "fontFace": "Cascadia Code PL"
    },
    "list": [
      {
        "commandline": "powershell.exe",
        "cursorColor": "#FFFFFD",
        "cursorShape": "filledBox",
        "experimental.retroTerminalEffect": false,
        "fontFace": "Cascadia Code PL",
        "guid": "{61c54bbd-c2c6-5271-96e7-009a87ff44bf}",
        "hidden": false,
        "name": "Windows PowerShell",
        "startingDirectory": null,
        "suppressApplicationTitle": true,
        "tabTitle": null,
        "useAcrylic": false
      },
}
```

Changing the `profiles` doesn't do anything but changing the font in list does change it.

PS: I can't access the nerdfonts.com site since I've probably been blocked by many sites last few months 😂 so apologies if u already have an answer there :)

**Comment by deadcoder0904** [comment]
Managed to finally make it work. I guess I did download Powerline Fonts but I didn't download it Patched. I now installed Windows Compatible fonts like https://github.com/ryanoasis/nerd-fonts/blob/master/patched-fonts/CascadiaCode/Regular/complete/Caskaydia%20Cove%20Regular%20Nerd%20Font%20Complete%20Windows%20Compatible.otf & changed the name accordingly.

```json
"list": [
  {
    "commandline": "powershell.exe",
    "cursorColor": "#FFFFFD",
    "cursorShape": "filledBox",
    "experimental.retroTerminalEffect": false,
    "fontFace": "CaskaydiaCove NF",
    "guid": "{61c54bbd-c2c6-5271-96e7-009a87ff44bf}",
    "hidden": false,
    "name": "Windows PowerShell",
    "startingDirectory": null,
    "suppressApplicationTitle": true,
    "tabTitle": null,
    "useAcrylic": false
  },
]
```

**Comment by patrickirvincw** [comment]
I had this problem - the default font was being overridden by List.
